specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp

```diff
@@ -3,75 +3,37 @@ import "@typespec/http";
 import "@typespec/versioning";
 
 import "./azure_chat_extensions.tsp";
+import "./azure_content_filtering.tsp";
+import "./chat_messages.tsp";
 import "./common.tsp";
 import "./functions.tsp";
+import "./tools.tsp";
 
 using TypeSpec.Rest;
 using TypeSpec.Http;
 using TypeSpec.Versioning;
 
 namespace Azure.OpenAI;
 
-@doc("A description of the intended purpose of a message within a chat completions interaction.")
-enum ChatRole {
-  @doc("The role that instructs or sets the behavior of the assistant.")
-  system: "system",
-
-  @doc("The role that provides responses to system-instructed, user-prompted input.")
-  assistant: "assistant",
-
-  @doc("The role that provides input for chat completions.")
-  user: "user",
-
-  @doc("The role that provides function results for chat completions.")
-  @added(ServiceApiVersions.v2023_07_01_Preview)
-  function: "function",
-
-  @doc("The role that represents extension tool activity within a chat completions operation.")
-  @added(ServiceApiVersions.v2023_08_01_Preview)
-  tool: "tool",
-}
-
-@doc("A single, role-attributed message within a chat completion interaction.")
-model ChatMessage {
```

Comment:
neither `ChatRole` nor `ChatMessage` need to be annotated with `@removed`? I understand that these are not retired models from the service but were just refactored. Is that why the annotation is not needed in this particular type of "removal"?

Reply:
`ChatRole` was moved into `common.tsp`, while `ChatMessage` was structurally replaced. There are "renamed" decorators and whatnot, but I'm not sure how we'd represent the decomposition replacement that happened with `ChatMessage` becoming a whole family of related things.